src/realm/util/string_buffer.hpp

```diff
@@ -132,16 +130,11 @@ inline const char* StringBuffer::data() const noexcept
     return m_buffer.data();
 }
 
-inline char* StringBuffer::c_str() noexcept
-{
-    char* d = data();
-    return d ? d : &m_zero;
-}
-
 inline const char* StringBuffer::c_str() const noexcept
 {
+    static const char zero = 0;
```

Comment:
I still believe this line is subject to a thread-safe check for every invocation. Without having access to the UWP environment myself, I cannot immediately suggest a better alternative, other than returning `""` and rely on the compiler being able to place such static, empty string properly (and only once) in a read-only section of the binary.